examples/erc1155-metadata-uri/src/lib.rs

```diff
@@ -0,0 +1,46 @@
+#![cfg_attr(not(test), no_main)]
+extern crate alloc;
+
+use alloy_primitives::{FixedBytes, U256};
+use openzeppelin_stylus::{
+    token::erc1155::{
+        extensions::{Erc1155MetadataUri, Erc1155UriStorage},
+        Erc1155,
+    },
+    utils::introspection::erc165::IErc165,
+};
+use stylus_sdk::prelude::{entrypoint, public, sol_storage};
+
+sol_storage! {
+    #[entrypoint]
+    struct Erc1155MetadataUriExample {
+        #[borrow]
+        Erc1155 erc1155;
+        #[borrow]
+        Erc1155MetadataUri metadata_uri;
+        Erc1155UriStorage uri_storage;
+    }
+}
+
+#[public]
+#[inherit(Erc1155, Erc1155MetadataUri)]
+impl Erc1155MetadataUriExample {
+    fn uri(&self, token_id: U256) -> String {
+        self.uri_storage.uri(token_id, &self.metadata_uri)
+    }
+
+    #[selector(name = "setTokenURI")]
+    fn set_token_uri(&mut self, token_id: U256, token_uri: String) {
+        self.uri_storage.set_token_uri(token_id, token_uri, &self.metadata_uri)
+    }
+
+    #[selector(name = "setBaseURI")]
+    fn set_base_uri(&mut self, base_uri: String) {
+        self.uri_storage.set_base_uri(base_uri)
+    }
+
+    fn supports_interface(interface_id: FixedBytes<4>) -> bool {
+        Erc1155::supports_interface(interface_id)
+            || Erc1155MetadataUri::supports_interface(interface_id)
```

Comment:
How about `Erc1155UriStorage`?

Reply:
Solidity version does not assert that this interface is supported, see:
- https://github.com/OpenZeppelin/openzeppelin-contracts/blob/6e05b68bd96ab90a4049adb89f20fd578404b274/contracts/token/ERC1155/ERC1155.sol
- https://github.com/OpenZeppelin/openzeppelin-contracts/blob/6e05b68bd96ab90a4049adb89f20fd578404b274/contracts/token/ERC1155/extensions/ERC1155URIStorage.sol

If we still wanted to implement this, we would need to create `IErc1155UriStorage` interface, annotate it with `#[interface_id]`, and implement it on the current struct. This would be the same interface as `IErc1155MetadataUri`.